layer/contracts/models.py

```diff
@@ -111,3 +114,15 @@ def set_model_runtime_objects(
     ) -> "Model":
         self._model_runtime_objects = model_runtime_objects
         return self
+
+    def add_log_data_runner(self, log_data_runner: LogDataRunner) -> None:
+        self._logged_data_runner = log_data_runner
+
+    def get_metadata(self, tag: str, step: Optional[int] = None) -> LoggedDataObject:
```

Comment:
Is this only meant for models, or datasets too? cc @volkangurel

Reply:
This is for datasets too.